instrumentation/reactor/reactor-3.1/library/src/test/java/io/opentelemetry/instrumentation/reactor/v3_1/ReactorCoreTest.java

```diff
@@ -406,6 +408,68 @@ void doesNotOverrideInnerCurrentSpansWithThereIsOuterCurrent() {
                         .hasAttributes(attributeEntry("onNext", true))));
   }
 
+  @Test
+  void doesNotLeakContextOnRetry() {
+    // retry calls subscribe again from onError where we have active context, check that this
+    // context is not used as parent for retried operations
+    AtomicBoolean beforeRetry = new AtomicBoolean(true);
+    Flux<Integer> publish =
+        Flux.create(
+            sink -> {
+              for (int i = 0; i < 2; i++) {
+                Span s =
+                    tracer
+                        .spanBuilder(
+                            "produce " + (beforeRetry.get() ? "before" : "after") + " retry " + i)
+                        .startSpan();
+                try (Scope scope = s.makeCurrent()) {
+                  sink.next(i);
+                } finally {
+                  s.end();
+                }
+              }
+            });
+
+    Flux.defer(() -> publish.delaySubscription(Duration.ofMillis(1)))
+        .doOnNext(
+            message -> {
+              tracer.spanBuilder("process " + message).startSpan().end();
+            })
+        .handle(
+            (message, sink) -> {
+              if (message == 1 && beforeRetry.compareAndSet(true, false)) {
+                sink.error(new RuntimeException("Message has error"));
+              } else {
+                sink.next(message);
+              }
+            })
+        .retry()
+        .subscribe();
```

Comment:
can you add a test where there's an active parent span when `.subscribe()` is called?

Reply:
I added a test with parent span, all child spans are under that parent.